### ISSUE TITLE
Cells  with a markdown horizontal line (---) get hidden

### ISSUE THREAD

**Issue by fperez** [executablebooks/jupyterlab-myst#56]
### Describe the bug

**context**

Any markdown cell that begins with a horizontal ruler marker is rendered invisible, as illustrated here:

![image](https://user-images.githubusercontent.com/57394/198944009-2c522e03-e5f6-4de1-a67d-52571e774476.png)

All it takes is making a markdown cell with

```
---
Any content
```

and once rendered, it gets displayed as a blank space as shown above.

Note: I originally reported this as a [JupyterLab bug](https://github.com/jupyterlab/jupyterlab/issues/13342), but @jangenoe's good eye figured out the issue was really with jupyterlab-myst. 

I happened to notice it when upgrading to 3.5.0, and got confused. A good reminder that "correlation is not causation" applies to software too :)

The bug is super easy to replicate, but if anyone is curious, I have screenshots and videos in that JLab report.

**expectation**

Md ruler is correctly rendered.

### Reproduce the bug

See above

### List your environment

I'm using jupyterlab-myst 0.1.6

**Comment by rowanc1** [comment]
Looking into this a bit -- it is expecting that this is front-matter, and treats it as a YAML block, that is likely not closed (or something like that), and the algorithm I think eats the content.

The fix for this needs to be turning off the frontmatter interpretation if we aren't in the first cell (and likely knowing that it is actually closed!)

**Comment by fperez** [comment]
Thx @rowanc1 - LMK if I can do anything else to help.

**Comment by fperez** [comment]
Checking in on whether now that the 2.0 release is out, how hard this one seems in the priority list... In our existing courses we have a _lot_ of usage of `---` to create horizontal rulers in our teaching notebooks, to clearly differentiate parts while teaching.

This bug makes it impossible to install the tools in our teaching hubs for those courses, unfortunately.

**Comment by agoose77** [comment]
Unless I'm mistaken one can use other delimeters for rules. Nevertheless we should fix this.

**Comment by fperez** [comment]
Agreed @agoose77 - the problem is that it's the most common one (and it's very intuitive too), and we have a _ton_ of content with them already in. I won't make any friends with my colleagues who have written those notebooks if I tell them to go change all their lines to satisfy MyST :) 

Thanks a lot for looking into it when you get a chance (and for all your great work on MyST!!).

**Comment by fperez** [comment]
Cc @ryanlovett.
